### PR TITLE
Nested folders: Add folder actions to other tabs

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- adds folder actions to the `Alerting` and `Panels` tabs
- creates new page components for these tabs. this is for a couple of reasons:
  - so they can use the rtk query method to get the folder information (same as `Dashboards` tab)
  - so they can build the navModel in a programmatic way (same as `Dashboards` tab)
- implements move/delete in the folder actions 

**Why do we need this feature?**

- better UX

**Who is this feature for?**

- anyone using nested folders!

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/browse-dashboards/BrowseFolderAlertingPage.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { useGetFolderQuery } from './api/browseDashboardsAPI';
+import { FolderActionsButton } from './components/FolderActionsButton';
+
+export interface OwnProps extends GrafanaRouteComponentProps<{ uid: string }> {}
+
+export function BrowseFolderAlertingPage({ match }: OwnProps) {
+  const { uid: folderUID } = match.params;
+  const { data: folderDTO, isLoading } = useGetFolderQuery(folderUID);
+  const folder = useSelector((state) => state.folder);
+
```
Comment (joshhunt) [inline public/app/features/browse-dashboards/BrowseFolderAlertingPage.tsx:19]: Can I be a pain and ask if we can write tests for this?

Something like

```ts
describe("BrowseFolderAlertingPage", () => {
    it("renders correct breadcrumbs"); 
    it("renders folder name");
    it("renders alerts for folder");
    it("shows the alerting tab as selected")
});
```

Likewise for the other one? 🥺

**Review by joshhunt — COMMENTED** [review]
Good with code, and gave it a spin and works well with nested folders enabled. I want to test as well when disabling the feature flag before giving it my ✔️

File: public/app/features/browse-dashboards/BrowseFolderAlertingPage.test.tsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    ...jest.requireActual('@grafana/runtime').config,
+    unifiedAlertingEnabled: true,
+  },
+}));
+
+const mockFolderName = 'myFolder';
+const mockFolderUid = '12345';
+const mockAlertRuleName = 'myAlertRule';
+
+const mockRulerRulesResponse: RulerRulesConfigDTO = {
```
Comment (joshhunt) [inline public/app/features/browse-dashboards/BrowseFolderAlertingPage.test.tsx:38]: I would like to discuss this with the wider frontend group - is there a better home where these fixtures can live? Maybe we could put a TODO on this to find a better home for it?

Personally, I would like to extend this fixtures pattern we/I started in browse-dashboards to the rest of the codebase...

**Review by joshhunt — APPROVED** [review]

File: public/app/features/browse-dashboards/BrowseFolderAlertingPage.test.tsx (hunk 2 above)
Comment (ashharrison90) [reply public/app/features/browse-dashboards/BrowseFolderAlertingPage.test.tsx]: have moved these into separate fixtures files, still in the `browse-dashboards` folder for now. should make it very easy to move them later (e.g. if we want to move the libraryElements fixtures to the libraryPanels folder)

**Pull request merged**